src/main/java/org/sonar/plugins/cas/ForceCasLoginFilter.java

```
@@ -106,15 +104,26 @@ int getMaxCookieAge(Configuration configuration) {
         return SonarCasProperties.URL_AFTER_CAS_REDIRECT_COOKIE_MAX_AGE_IN_SECS.getInteger(configuration, DEFAULT_CAS_REDIRECT_COOKIE_AGE);
     }
 
-    private void redirectToLogin(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    private void redirectToLogin(HttpRequest request, HttpResponse response) throws IOException {
         LOG.debug("Redirecting to login page {} -> {}", request.getRequestURL(), SONAR_LOGIN_URL_PATH);
 
         response.sendRedirect(request.getContextPath() + SONAR_LOGIN_URL_PATH);
     }
 
-    private boolean isAuthenticated(HttpServletRequest request) {
+    private boolean isAuthenticated(HttpRequest request) {
         // https://github.com/SonarSource/sonarqube/blob/9973bacbfa4a945e509bf1b574d7e5aae4ba155a/server/sonar-server/src/main/java/org/sonar/server/authentication/UserSessionInitializer.java#L138
-        String login = StringUtils.defaultString((String) request.getAttribute("LOGIN"));
+        Object delegate;
+        String login = "";
+        try {
+            Method getDelegate = request.getClass().getMethod("getDelegate");
+            delegate = getDelegate.invoke(request);
+
+            Method getAttribute  = delegate.getClass().getMethod("getAttribute", String.class);
+            login = (String) getAttribute.invoke(delegate, "LOGIN");
+
+        } catch (Exception e) {
+            // ignore the exception
```

why can we ignore the exception here?